tests/unit/htdocs/class/XoopsUserUtilityTest.php

```
@@ -0,0 +1,508 @@
+    /**
+     * HTTP_X_FORWARDED_FOR should be used when present.
+     */
+    public function testReadsHttpXForwardedFor(): void
+    {
+        $_SERVER['REMOTE_ADDR'] = '127.0.0.1';
+        $_SERVER['HTTP_X_FORWARDED_FOR'] = '192.168.1.10';
+
+        $result = \XoopsUserUtility::getIP(true);
+
+        $this->assertSame('192.168.1.10', $result);
+    }
```

PHPMD (2.15.0)

[error] testReadsHttpXForwardedFor accesses the super-global variable $_SERVER. (undefined)

(Superglobals)